### PR TITLE
dev-db/cpp-driver: Add ebuild of Cassandra C/C++ driver

### PR DESCRIPTION
**Pull request by geaaru**
New ebuild / package for Cassandra C/C++ driver.

### CODE REVIEW

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: move to `EAPI=6` please

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+RESTRICT="nomirror"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-libs/libuv
+	dev-libs/openssl
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: Please try whether this package supports **libressl**, as Gentoo wants to provide **libressl** and **openssl** across the whole tree. See https://wiki.gentoo.org/wiki/Project:LibreSSL

In addition, openssl will require (sub)-slot operators `:0=`

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+"
+DEPEND="
+	dev-util/cmake
+${RDEPEND}
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: by `inherit cmake-utils` you already implictly depend on cmake. Change this to `DEPEND="${RDEPEND}"`

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-libs/libuv
+	|| (
+		dev-libs/openssl:0
+		dev-libs/libressl
+	)
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: this is incorrect, please read how this should be done on the project page: https://wiki.gentoo.org/wiki/Project:LibreSSL

You'll need to add `libressl` to `IUSE` and then in `RDEPEND` do

```
!libressl? ( dev-libs/openssl:0= )
libressl? ( dev-libs/libressl:0= )
```

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="libressl"
+
+RDEPEND="
+	dev-libs/libuv
+	!libressl? ( dev-libs/openssl:0= )
+	libressl? ( dev-libs/libressl:0= )
+"
+DEPEND="
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: simplify this to `DEPEND="${RDEPEND}"`

**Comment by SoapGentoo** [comment]
@geaaru have you tested libressl support? Just adding it is not enough to ensure that it works properly. Furthermore, your submission still has a few problems:

Your **COMMITMSG**s are not in compliance with Gentoo's **COMMITMSG**s. The easiest way to make them compliant is to always use repoman (where you should start adding a GPG key for the future).

Second, you should always squash commits into one logical commit that does one task. Have a look here: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by geaaru** [comment]
Ok for rebase.
About COMMITMSG, said me what message I must use on rebase, I will see how use repoman in the future. I haven't time now. sorry.
About libressl, I can't currently test with libressl. If you think that doesn't work could be removed or we can leave libressl use flag and see will be received bugs about this in bug tracking system.

Bye

**Comment by doughdemon** [comment]
The upstream webpage says that openssl is optional. If that is so, an ssl USE flag should be added. The same page says that testing has additional requirements. Add these dependencies and put them behind a test use flag. (If the tests work. If they don't, it might be better to add RESTRICT=test in.)

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	local mycmakeargs=()
+
+	if ! use ssl ; then
+		mycmakeargs=(
+			-DCASS_USE_OPENSSL:BOOL=OFF
+		)
+	fi
+
+	cmake-utils_src_configure
+}
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: simplify `src_configure()` to:

```
src_configure() {
	local mycmakeargs=( -DCASS_USE_OPENSSL=$(usex ssl) )
	cmake-utils_src_configure
}
```

File: dev-db/cpp-driver/cpp-driver-2.4.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+	dev-libs/libuv
+	ssl? (
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl:0= )
+	)
+"
+DEPEND="${RDEPEND}"
+
+src_configure() {
+	local mycmakeargs=( -DCASS_USE_OPENSSL:BOOL=$(usex ssl) )
```
Comment (SoapGentoo) [inline dev-db/cpp-driver/cpp-driver-2.4.3.ebuild]: remove the `:BOOL` specifier, it's optional and not needed in Gentoo

**Comment by doughdemon** [comment]
I've just tested building with libressl and it seems to work.

**Comment by SoapGentoo** [comment]
@doughdemon thanks for testing, I'll merge it momentarily.

**Pull request closed without merging**